File: leetcode/c++/lc189.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int len = static_cast<int>(nums.size());
        k = k%len;
        vector<int> temp;
        for(int i = len - k; i < len; i++)  {
            temp.push_back(nums[i]);
        }
        for(int i = 0; i < len - k; i++) {
            temp.push_back(nums[i]);
        }
        nums = temp;

        for(int n : nums) {
            cout << n << ", ";
        }      
    }
};

int main() {
    Solution sol;
    vector<int> nums = {-1};
    sol.rotate(nums, 3);
}

File: leetcode/python/lc189.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        k %= n
        aux = nums[-k:]
        aux += nums[: n - k]
        for i in range(n):
            nums[i] = aux[i]


nums = [1, 2, 3, 4, 5, 6]
k = 1
s = Solution()
s.rotate(nums, k)
for i in nums:
    print(i)
